web3/main.py

```diff
@@ -180,5 +180,7 @@ def twos_compliment_of_negative(value, bit_size):
     return remove_0x_prefix(hex_value)
 
 
-def string_to_hex(string):
-    return codecs.getencoder('hex')(string)[0]
+def bytes_to_hex(byte_string):
+    if type(byte_string) == str:
+        byte_string = str.encode(byte_string)
+    return codecs.getencoder('hex')(byte_string)[0].decode("utf-8")
```

Comment:
Explicit encoding is better than implicit here, like `byte_string.encode('utf-8')` instead of `str.encode(byte_string)`

You could also leverage some of the tools from the utlis library here, like: `return remove_0x_prefix(to_hex(byte_string))`